db/migrate/20150228200007_split_annotations.rb

```diff
@@ -0,0 +1,43 @@
+class SplitAnnotations < ActiveRecord::Migration
+  def change
+
+    Annotation.find_each do |annotation|
+        text = annotation.text
+
+
+        res = text.split("[");
+        
+        if res.size == 1 then
+            annotation.comment = text
+        end
+
+        if res.size > 1 then
+
+            len = res.size
+            points = res[len-1]
+            res.delete_at(len-1)
+
+            points = points.split(":");
+            comment = res.join("[");
+            annotation.comment = comment
+           
+            if points.size == 1 then
+              annotation.value = points[0].delete("]").to_f
+            else
+              if points[0] == "?" then
+                annotation.problem = points[1].delete("]").to_i
```

Comment:
You're right. e.g. "don't leave in print statements [-0.5:CancelFlightsCorrectly]"

But we switched to IDs as of Monday...

I think I'll try to parseInt and if it fails, it means that it's just the name of the problem so I'll find by name. How does that sounds?